sentry-log4j2/src/main/java/io/sentry/log4j2/SentryAppender.java

```diff
@@ -0,0 +1,220 @@
+package io.sentry.log4j2;
+
+import io.sentry.core.Breadcrumb;
+import io.sentry.core.DateUtils;
+import io.sentry.core.Sentry;
+import io.sentry.core.SentryEvent;
+import io.sentry.core.SentryLevel;
+import io.sentry.core.SentryOptions;
+import io.sentry.core.protocol.Message;
+import io.sentry.core.protocol.SdkVersion;
+import io.sentry.core.transport.ITransport;
+import io.sentry.core.util.CollectionUtils;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.logging.log4j.Level;
+import org.apache.logging.log4j.core.Filter;
+import org.apache.logging.log4j.core.LogEvent;
+import org.apache.logging.log4j.core.appender.AbstractAppender;
+import org.apache.logging.log4j.core.config.plugins.Plugin;
+import org.apache.logging.log4j.core.config.plugins.PluginAttribute;
+import org.apache.logging.log4j.core.config.plugins.PluginElement;
+import org.apache.logging.log4j.core.config.plugins.PluginFactory;
+import org.apache.logging.log4j.core.impl.ThrowableProxy;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+/** Appender for Log4j2 in charge of sending the logged events to a Sentry server. */
+@Plugin(name = "Sentry", category = "Core", elementType = "appender", printObject = true)
+public final class SentryAppender extends AbstractAppender {
+  private @Nullable String dsn;
+  private @Nullable ITransport transport;
+  private @NotNull Level minimumBreadcrumbLevel = Level.INFO;
+  private @NotNull Level minimumEventLevel = Level.ERROR;
+
+  public SentryAppender(
```

Comment:
we could have a ctor overload that takes only the necessary, so its not needed to pass  minimumBreadcrumbLevel and minimumEventLevel, its gonna be the default values, unless this is not instantiated manually by clients, so then it doesn't matter

Reply:
it's not instantiated manually by users. Users use `log4j2.xml` to configure appenders.